pages/plants/[id]/index.js

```diff
@@ -0,0 +1,114 @@
+import BackButton from "@/components/BackButton";
+
+import Image from "next/image";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import useSWR from "swr";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const { isReady } = router;
+  const { id } = router.query;
+
+  const { data: plant, isLoading, error } = useSWR(`/api/plants/${id}`);
+
+  if (!isReady || isLoading || error || !plant) return <h2>Loading...</h2>;
```

Comment:
please split the "loading" and "error" cases into two if statements with different return values.